xrpl4j-client/src/main/java/org/xrpl/xrpl4j/client/Finality.java

```diff
@@ -0,0 +1,31 @@
+package org.xrpl.xrpl4j.client;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.immutables.value.Value;
+
+import java.util.Optional;
+
+@Value.Immutable
+@JsonSerialize(as = ImmutableFinality.class)
+@JsonDeserialize(as = ImmutableFinality.class)
+public interface Finality {
+
+  /**
+   * Construct a builder for this class.
+   *
+   * @return An {@link ImmutableFinality.Builder}.
+   */
+  static ImmutableFinality.Builder builder() {
+    return ImmutableFinality.builder();
+  }
+
+  XrplClient.FinalityStatus finalityStatus();
+
+  Optional<String> resultCode();
+
+  @Value.Auxiliary
```

Comment:
Any particular reason this is marked `@Auxilary`?

Reply:
It cannot be assigned a default value and auxiliary will remove the attribute from equals comparison. But now that I run without any annotation, I feel like there is no need for one. I removed it!

Reply:
Oh, my bad -- I didn't realize this was a `...Safe` variant, so you were correct to mark it as `Auxiliary`. The _actual_ value for equals/etc will be taken from `resultCode()`.

Please add this annotation back -- sorry for leading you in the wrong direction.

Reply:
I think `resultCode` is the proper name. If you look [here](https://xrpl.org/transaction-results.html#transaction-results), there are two types of result: (1) and `engine_result` and (2) a transaction result defined in metadata. To me calling it `resultCode` seems like it captures both meanings.

That said, @nkramer44 WDYT? Should we call this `engineResult` instead?

Reply:
I wouldn't call it `engineResult` personally. That is the field you get back in a `SubmitResult`. While the values of that field are the same ones you see in the `TransactionMetadata` field, I think it's clearer to call it `resultCode`, since this is what you see in `TransactionMetadata`

Reply:
Okay!